carbon-design-system/carbon
# fix(menu): render through portal to avoid z-index and overflow issues

Renders the `Menu` component through a portal to avoid it being misplaced, cut off or invisible in certain setups (such as the DataTable toolbar).

#### Changelog

**Changed**

- render `Menu` through `ReactDOM.createPortal()`

#### Testing / Reviewing

- Ensure keyboard navigation still works as expected
- Ensure the position of the menu is corretl attached to the trigger point
- Ensure that when the menu is closed through `blur`, the focus is returned to the trigger element (if any exists)
  - I added a test story "unstable_menu > OverflowMenu > In Toolbar" to test this.
  - **The story should be removed prior to merging and only serves as an example setup to illustrate the purpose of this PR**
